src/game/logic/ai/aipathfind.h

```diff
@@ -57,13 +82,154 @@ class Path : public MemoryPoolObject, public SnapShot
     PathNode *m_unkNode;
 };
 
-class Pathfinder
+class PathfindLayer
+{
+public:
+    PathfindLayer();
+    ~PathfindLayer();
+    void Reset();
+
+private:
+    PathfindCell *m_map; // not 100% confirmed
+    PathfindCell **m_mapPointers; // not 100% confirmed
+    IRegion2D m_extent; // not 100% confirmed
+    IRegion2D m_unkRegion; // not 100% confirmed
+    PathfindLayerEnum m_layerNum; // not 100% confirmed
+    int m_unk; // not 100% confirmed
+    Bridge *m_bridge; // not 100% confirmed
+    bool m_destroyed; // not 100% confirmed
+};
+
+class PathfindServicesInterface
+{
+public:
+    virtual Path *Find_Path(Object *obj, const LocomotorSet &locomotor_set, const Coord3D *from, const Coord3D *raw_to) = 0;
+    virtual Path *Find_Closest_Path(Object *obj,
+        const LocomotorSet &locomotor_set,
+        const Coord3D *from,
+        Coord3D *raw_to,
+        bool blocked,
+        float path_cost_multiplier,
+        bool move_allies) = 0;
+    virtual Path *Find_Attack_Path(const Object *obj,
+        const LocomotorSet &locomotor_set,
+        const Coord3D *from,
+        const Object *victim,
+        const Coord3D *victim_pos,
+        const Weapon *weapon) = 0;
+    virtual Path *Patch_Path(const Object *obj, const LocomotorSet &locomotor_set, Path *path, bool blocked) = 0;
+    virtual Path *Find_Safe_Path(const Object *obj,
+        const LocomotorSet &locomotor_set,
+        const Coord3D *pos1,
+        const Coord3D *pos2,
+        const Coord3D *pos3,
+        float f) = 0;
+};
+
+class PathfindCellInfo
 {
 public:

```

Comment:
Constructor will be added later?

Reply:
There is no constructor for PathfindCellInfo at all (none exists even where you would expect one in ZH WB)